galaxy_ng/app/tasks/publishing.py

```diff
@@ -34,29 +34,23 @@ def get_created_collection_versions():
     return created_collection_versions
 
 
-def import_and_move_to_staging(temp_file_pk, **kwargs):
+def import_and_move_to_staging(username, repository_pk=None, **kwargs):
     """Import collection version and move to staging repository.
 
-    Custom task to call pulp_ansible's import_collection() task then
+    Custom task to call pulpcore's general_create() task then
     enqueue two tasks to add to staging repo and remove from inbound repo.
 
     This task will not wait for the enqueued tasks to finish.
     """
-    inbound_repository_pk = kwargs.get('repository_pk')
-    import_collection(
-        temp_file_pk=temp_file_pk,
-        repository_pk=inbound_repository_pk,
-        expected_namespace=kwargs['expected_namespace'],
-        expected_name=kwargs['expected_name'],
-        expected_version=kwargs['expected_version'],
-    )
+    general_args = kwargs.pop("general_args")
```

Comment:
should there be a similar line `repository_pk = kwargs.pop("repository_pk")`

or does `dispatch` extract `repository_pk` from kwargs in `dispatch(import_and_move_to_staging, exclusive_resources=locks, kwargs=kwargs)` and pass here as a named param?

Reply:
The latter, `repository_pk` gets extracted as a named param.